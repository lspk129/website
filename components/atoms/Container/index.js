import styled from 'styled-components';

const Container = styled.div`
  ${({ display }) => display && `display: ${display}`}
  ${({ textAlign }) => textAlign && `text-align: ${textAlign}`};
  ${({ alignSelf }) => alignSelf && `align-self: ${alignSelf}`};
  ${({ alignItems }) => alignItems && `align-items: ${alignItems}`};

  /* margin properties */
  ${({ m }) => m && `margin: ${m}rem;`}
  ${({ mx }) => mx && `margin: 0rem ${mx}rem;`}
  ${({ my }) => my && `margin: ${my}rem 0rem;`}
  ${({ mt }) => mt && `margin-top: ${mt}rem;`}
  ${({ mb }) => mb && `margin-bottom: ${mb}rem;`}
  ${({ mr }) => mr && `margin-right: ${mr}rem;`}
  ${({ ml }) => ml && `margin-left: ${ml}rem;`}

  /* padding properties */
  ${({ p }) => p && `padding: ${p}rem;`}
  ${({ px }) => px && `padding: 0rem ${px}rem;`}
  ${({ py }) => py && `padding: ${py}rem 0rem;`}
  ${({ pt }) => pt && `padding-top: ${pt}rem;`}
  ${({ pb }) => pb && `padding-bottom: ${pb}rem;`}
  ${({ pr }) => pr && `padding-right: ${pr}rem;`}
  ${({ pl }) => pl && `padding-left: ${pl}rem;`}

  /* background properties */
  ${({ bg, src }) => bg && `
    background-image: url(${src});
    background-repeat: no-repeat;
    background-attachment: fixed;
    background-size: cover;
    display: grid;
  `}
  ${({ backgroundColor }) => backgroundColor && `
    background-color: ${backgroundColor};
  `}
  ${({ zIndex }) => zIndex && `z-index: ${zIndex};`}
`;

export default Container;
