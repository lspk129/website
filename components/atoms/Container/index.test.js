import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Container from './';

describe('Container', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Container />);
    expect(wrapper).toBePresent();
    expect(wrapper).toHaveTagName('div');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('render with display props', () => {
    const wrapper = mount(<Container display={'grid'} />);
    expect(wrapper).toHaveProp('display', 'grid');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('render with textAlign props', () => {
    const wrapper = mount(<Container textAlign={'center'} />);
    expect(wrapper).toHaveProp('textAlign', 'center');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('render with alignSelf props', () => {
    const wrapper = mount(<Container alignSelf={'flex-end'} />);
    expect(wrapper).toHaveProp('alignSelf', 'flex-end');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('render with alignItems props', () => {
    const wrapper = mount(<Container alignItems={'flex-end'} />);
    expect(wrapper).toHaveProp('alignItems', 'flex-end');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('render with margin props', () => {
    const wrapper = mount(<Container mx={'1'} />);
    expect(wrapper).toHaveProp('mx', '1');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('render with margin left and right props', () => {
    const wrapper = mount(<Container mx={2} />);
    expect(wrapper).toHaveProp('mx', 2);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('render with margin top and bottom props', () => {
    const wrapper = mount(<Container my={3} />);
    expect(wrapper).toHaveProp('my', 3);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('render with margin top props', () => {
    const wrapper = mount(<Container mt={4} />);
    expect(wrapper).toHaveProp('mt', 4);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with margin bottom props', () => {
    const wrapper = mount(<Container mb={5} />);
    expect(wrapper).toHaveProp('mb', 5);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with margin right props', () => {
    const wrapper = mount(<Container mr={6} />);
    expect(wrapper).toHaveProp('mr', 6);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with margin left props', () => {
    const wrapper = mount(<Container ml={7} />);
    expect(wrapper).toHaveProp('ml', 7);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with padding props', () => {
    const wrapper = mount(<Container p={8} />);
    expect(wrapper).toHaveProp('p', 8);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Container with padding left and right props', () => {
    const wrapper = mount(<Container px={7} />);
    expect(wrapper).toHaveProp('px', 7);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with padding top and bottom props', () => {
    const wrapper = mount(<Container py={6} />);
    expect(wrapper).toHaveProp('py', 6);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with padding top props', () => {
    const wrapper = mount(<Container pt={5} />);
    expect(wrapper).toHaveProp('pt', 5);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with padding bottom props', () => {
    const wrapper = mount(<Container pb={4} />);
    expect(wrapper).toHaveProp('pb', 4);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with padding right props', () => {
    const wrapper = mount(<Container pr={3} />);
    expect(wrapper).toHaveProp('pr', 3);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with padding left props', () => {
    const wrapper = mount(<Container pl={2} />);
    expect(wrapper).toHaveProp('pl', 2);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with background color props', () => {
    const wrapper = mount(<Container backgroundColor={'#BADA55'} />);
    expect(wrapper).toHaveProp('backgroundColor', '#BADA55');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with background image source props', () => {
    const wrapper = mount(<Container src={'../../../static/Service_image.jpg'} />);
    expect(wrapper).toHaveProp('src', '../../../static/Service_image.jpg');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with z-index props', () => {
    const wrapper = mount(<Container zIndex={10} />);
    expect(wrapper).toHaveProp('zIndex', 10);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
