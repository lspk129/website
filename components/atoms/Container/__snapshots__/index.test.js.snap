// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container render with alignItems props 1`] = `
<styled.div
  alignItems="flex-end"
>
  <div
    className="sc-bdVaJa fYwkxx"
  />
</styled.div>
`;

exports[`Container render with alignSelf props 1`] = `
<styled.div
  alignSelf="flex-end"
>
  <div
    className="sc-bdVaJa cCdSpl"
  />
</styled.div>
`;

exports[`Container render with display props 1`] = `
<styled.div
  display="grid"
>
  <div
    className="sc-bdVaJa cFhjwl"
    display="grid"
  />
</styled.div>
`;

exports[`Container render with margin left and right props 1`] = `
<styled.div
  mx={2}
>
  <div
    className="sc-bdVaJa fbQCki"
  />
</styled.div>
`;

exports[`Container render with margin props 1`] = `
<styled.div
  mx="1"
>
  <div
    className="sc-bdVaJa huvOTw"
  />
</styled.div>
`;

exports[`Container render with margin top and bottom props 1`] = `
<styled.div
  my={3}
>
  <div
    className="sc-bdVaJa hJXbgd"
  />
</styled.div>
`;

exports[`Container render with margin top props 1`] = `
<styled.div
  mt={4}
>
  <div
    className="sc-bdVaJa aGRRn"
  />
</styled.div>
`;

exports[`Container render with textAlign props 1`] = `
<styled.div
  textAlign="center"
>
  <div
    className="sc-bdVaJa eFKGyX"
  />
</styled.div>
`;

exports[`Container renders Container with padding left and right props 1`] = `
<styled.div
  px={7}
>
  <div
    className="sc-bdVaJa hWWkBz"
  />
</styled.div>
`;

exports[`Container renders correctly 1`] = `
<div
  className="sc-bdVaJa bYXxhh"
/>
`;

exports[`Container renders with background color props 1`] = `
<styled.div
  backgroundColor="#BADA55"
>
  <div
    className="sc-bdVaJa fObLte"
  />
</styled.div>
`;

exports[`Container renders with background image source props 1`] = `
<styled.div
  src="../../../static/Service_image.jpg"
>
  <div
    className="sc-bdVaJa bYXxhh"
    src="../../../static/Service_image.jpg"
  />
</styled.div>
`;

exports[`Container renders with margin bottom props 1`] = `
<styled.div
  mb={5}
>
  <div
    className="sc-bdVaJa kSycwZ"
  />
</styled.div>
`;

exports[`Container renders with margin left props 1`] = `
<styled.div
  ml={7}
>
  <div
    className="sc-bdVaJa gkjOyO"
  />
</styled.div>
`;

exports[`Container renders with margin right props 1`] = `
<styled.div
  mr={6}
>
  <div
    className="sc-bdVaJa ldBSLb"
  />
</styled.div>
`;

exports[`Container renders with padding bottom props 1`] = `
<styled.div
  pb={4}
>
  <div
    className="sc-bdVaJa ePBnOn"
  />
</styled.div>
`;

exports[`Container renders with padding left props 1`] = `
<styled.div
  pl={2}
>
  <div
    className="sc-bdVaJa fqGAhs"
  />
</styled.div>
`;

exports[`Container renders with padding props 1`] = `
<styled.div
  p={8}
>
  <div
    className="sc-bdVaJa lgLEEq"
  />
</styled.div>
`;

exports[`Container renders with padding right props 1`] = `
<styled.div
  pr={3}
>
  <div
    className="sc-bdVaJa kGKvUv"
  />
</styled.div>
`;

exports[`Container renders with padding top and bottom props 1`] = `
<styled.div
  py={6}
>
  <div
    className="sc-bdVaJa VSVdp"
  />
</styled.div>
`;

exports[`Container renders with padding top props 1`] = `
<styled.div
  pt={5}
>
  <div
    className="sc-bdVaJa kJdgig"
  />
</styled.div>
`;

exports[`Container renders with z-index props 1`] = `
<styled.div
  zIndex={10}
>
  <div
    className="sc-bdVaJa coDzJd"
  />
</styled.div>
`;
